src/test/kotlin/com/atlassian/performance/tools/report/jfr/JfrFilterTest.kt

```diff
@@ -0,0 +1,137 @@
+package com.atlassian.performance.tools.report.jfr
+
+import com.atlassian.performance.tools.report.api.jfr.JfrFilter
+import com.atlassian.performance.tools.report.api.result.CompressedResult
+import jdk.jfr.consumer.RecordedEvent
+import org.apache.logging.log4j.LogManager
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.openjdk.jmc.flightrecorder.testutils.parser.*
+import java.io.File
+import java.io.FileInputStream
+import java.nio.file.Path
+import java.util.function.Predicate
+
+class JfrFilterTest {
+    private val logger = LogManager.getLogger(this::class.java)
+    private val zippedInput = File(javaClass.getResource("/profiler-result.zip")!!.toURI())
+
+    data class Chunk(
+        val eventsCount: Map<Long, Long>,
+        val header: ChunkHeader,
+        val metadataEvents: List<MetadataEvent>,
+        val eventTypes: List<Long>,
+        val eventSizes: List<Long>
+    ) {
+        override fun toString(): String {
+            return "Chunk(eventsCount=$eventsCount, header=$header, metadataEvent=$metadataEvents)"
+        }
+    }
+
+
+    private fun Path.summary(): List<Chunk> {
+        val eventsCount = mutableMapOf<Long, Long>()
+        var chunkHeader: ChunkHeader? = null
+        val metadataEvents = mutableListOf<MetadataEvent>()
+        val eventTypes = mutableListOf<Long>()
+        val eventSizes = mutableListOf<Long>()
+        val result = mutableListOf<Chunk>()
+        FileInputStream(this.toFile()).use {
```

Comment:
This stream is not used. The `.parse(this)` is opening its own stream underneath.